config/configunmarshaler/testdata/valid-config.yaml

```diff
@@ -20,6 +20,11 @@ extensions:
     extra: "some string"
 
 service:
+  telemetry:
```

Comment:
Would you be able to add an example of what's expected to happen when this is used, and what's expected when "json" encoding is used? I tried this PR here locally, they look the same:

```
extensions:
  zpages:

receivers:
  otlp:
    protocols:
      grpc:

processors:

exporters:
  logging:

service:
  telemetry:
    logs:
      level: "DEBUG"
      development: true
      encoding: "json"
  extensions: [zpages]
  pipelines:
    traces:
      receivers:
        - otlp
      processors: []
      exporters:
        - logging
```

With this output:

```
$ go run ./cmd/otelcol/ --config local/simple.yaml 
2021-09-15T10:19:06.664+0200	info	service/collector.go:255	Starting otelcol...	{"Version": "latest", "NumCPU": 8}
2021-09-15T10:19:06.665+0200	info	builder/exporters_builder.go:266	Exporter was built.	{"kind": "exporter", "name": "logging"}
2021-09-15T10:19:06.665+0200	info	builder/pipelines_builder.go:216	Pipeline was built.	{"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-09-15T10:19:06.665+0200	info	builder/receivers_builder.go:229	Receiver was built.	{"kind": "receiver", "name": "otlp", "datatype": "traces"}
2021-09-15T10:19:06.665+0200	info	service/service.go:93	Starting extensions...
2021-09-15T10:19:06.665+0200	info	builder/extensions_builder.go:54	Extension is starting...	{"kind": "extension", "name": "zpages"}
2021-09-15T10:19:06.665+0200	info	zpagesextension/zpagesextension.go:40	Register Host's zPages	{"kind": "extension", "name": "zpages"}
2021-09-15T10:19:06.666+0200	info	zpagesextension/zpagesextension.go:53	Starting zPages extension	{"kind": "extension", "name": "zpages", "config": {"TCPAddr":{"Endpoint":"localhost:55679"}}}
2021-09-15T10:19:06.666+0200	info	builder/extensions_builder.go:60	Extension started.	{"kind": "extension", "name": "zpages"}
2021-09-15T10:19:06.666+0200	info	service/service.go:98	Starting exporters...
2021-09-15T10:19:06.666+0200	info	builder/exporters_builder.go:93	Exporter is starting...	{"kind": "exporter", "name": "logging"}
2021-09-15T10:19:06.666+0200	info	builder/exporters_builder.go:98	Exporter started.	{"kind": "exporter", "name": "logging"}
2021-09-15T10:19:06.666+0200	info	service/service.go:103	Starting processors...
2021-09-15T10:19:06.666+0200	info	builder/pipelines_builder.go:52	Pipeline is starting...	{"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-09-15T10:19:06.666+0200	info	builder/pipelines_builder.go:63	Pipeline is started.	{"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-09-15T10:19:06.666+0200	info	service/service.go:108	Starting receivers...
2021-09-15T10:19:06.666+0200	info	builder/receivers_builder.go:71	Receiver is starting...	{"kind": "receiver", "name": "otlp"}
2021-09-15T10:19:06.666+0200	info	otlpreceiver/otlp.go:74	Starting GRPC server on endpoint 0.0.0.0:4317	{"kind": "receiver", "name": "otlp"}
2021-09-15T10:19:06.666+0200	info	builder/receivers_builder.go:76	Receiver started.	{"kind": "receiver", "name": "otlp"}
2021-09-15T10:19:06.666+0200	info	service/telemetry.go:65	Setting up own telemetry...
2021-09-15T10:19:06.679+0200	info	service/telemetry.go:113	Serving Prometheus metrics	{"address": ":8888", "level": 0, "service.instance.id": "650fdd8a-9b38-4e1f-a982-2646b651c7c0"}
2021-09-15T10:19:06.679+0200	info	service/collector.go:177	Everything is ready. Begin running and processing data.
^X^C2021-09-15T10:19:09.616+0200	info	service/collector.go:191	Received signal from OS	{"signal": "interrupt"}
2021-09-15T10:19:09.616+0200	info	service/collector.go:278	Starting shutdown...
2021-09-15T10:19:09.616+0200	info	service/service.go:128	Stopping receivers...
2021-09-15T10:19:09.616+0200	info	service/service.go:133	Stopping processors...
2021-09-15T10:19:09.616+0200	info	builder/pipelines_builder.go:71	Pipeline is shutting down...	{"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-09-15T10:19:09.616+0200	info	builder/pipelines_builder.go:77	Pipeline is shutdown.	{"pipeline_name": "traces", "pipeline_datatype": "traces"}
2021-09-15T10:19:09.616+0200	info	service/service.go:138	Stopping exporters...
2021-09-15T10:19:09.616+0200	info	service/service.go:143	Stopping extensions...
2021-09-15T10:19:09.616+0200	info	service/collector.go:294	Shutdown complete.
```

Reply:
IIUC that will happen on #4011